src/openms/source/FORMAT/QuantmsIO.cpp

```diff
@@ -0,0 +1,450 @@
+      for (size_t i = 0; i < peptide_evidences.size(); ++i)
+      {
+        if (i > 0) protein_accessions += ",";
+        protein_accessions += peptide_evidences[i].getProteinAccession();
+      }
+      status = mp_accessions_builder.Append(protein_accessions.c_str());
+      if (!status.ok()) {
+        throw OpenMS::Exception::ConversionError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, 
+                                        "Failed to append protein accessions: " + OpenMS::String(status.ToString()));
+      }
+
+      // Predicted RT (null for now)
+      status = predicted_rt_builder.AppendNull();
+      if (!status.ok()) {
+        throw OpenMS::Exception::ConversionError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, 
+                                        "Failed to append predicted rt: " + OpenMS::String(status.ToString()));
+      }
+
+      // Reference file name
+      OpenMS::String file_name = peptide_id.getSpectrumReference();
+      if (file_name.empty())
+      {
+        file_name = peptide_id.getBaseName();
+      }
+      if (file_name.empty())
+      {
+        file_name = "unknown";
+      }
+      status = reference_file_name_builder.Append(file_name.c_str());
+      if (!status.ok()) {
+        throw OpenMS::Exception::ConversionError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, 
+                                        "Failed to append reference file name: " + OpenMS::String(status.ToString()));
+      }
+
+      // Scan
+      OpenMS::String scan = peptide_id.getSpectrumReference();
+      if (scan.empty())
+      {
+        // Generate scan from RT if available
+        std::ostringstream scan_stream;
+        scan_stream << "RT_" << peptide_id.getRT();
+        scan = scan_stream.str();
+      }
+      status = scan_builder.Append(scan.c_str());
+      if (!status.ok()) {
+        throw OpenMS::Exception::ConversionError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, 
+                                        "Failed to append scan: " + OpenMS::String(status.ToString()));
+      }
+
+      // RT (retention time)
+      double rt = peptide_id.getRT();
+      if (rt >= 0)
+      {
+        status = rt_builder.Append(static_cast<float>(rt));
+      }
+      else
+      {
+        status = rt_builder.AppendNull();
+      }
+      if (!status.ok()) {
+        throw OpenMS::Exception::ConversionError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, 
+                                        "Failed to append rt: " + OpenMS::String(status.ToString()));
+      }
+
+      // Ion mobility (null for now)
+      status = ion_mobility_builder.AppendNull();
+      if (!status.ok()) {
+        throw OpenMS::Exception::ConversionError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, 
+                                        "Failed to append ion mobility: " + OpenMS::String(status.ToString()));
+      }
+
+      // Num peaks (null for now)
+      status = num_peaks_builder.AppendNull();
```

Comment:
TODO: add a new constant to Constants.h
if metaValueExist retrieve and set it with getMetaValue